x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix_vega_lite_spec.ts

```diff
@@ -0,0 +1,194 @@
+    // for repeated charts, it seems to be fixed for facets but not repeat.
+    // This causes #ddd lines to stand out in dark mode.
+    // See: https://github.com/vega/vega-lite/issues/5908
+    view: { fill: 'transparent', stroke: euiTheme.euiColorLightestShade },
+    padding: 10,
+    config: {
+      axis: {
+        domainColor: euiTheme.euiColorLightShade,
+        gridColor: euiTheme.euiColorLightestShade,
+        tickColor: euiTheme.euiColorLightestShade,
+        labelColor: euiTheme.euiTextSubduedColor,
+        titleColor: euiTheme.euiTextSubduedColor,
+      },
+    },
+    repeat: {
+      column: columns,
+      row: columns.slice().reverse(),
+    },
+    spec: {
+      data: { values },
+      mark: {
+        ...(outliers && dynamicSize
+          ? {
+              type: 'circle',
+              strokeWidth: 1.2,
+              strokeOpacity: 0.75,
+              fillOpacity: 0.1,
+            }
+          : { type: 'circle', opacity: 0.75, size: 8 }),
+      },
+      encoding: {
+        color: getColorSpec(euiTheme, outliers, color, legendType),
+        ...(dynamicSize
+          ? {
+              stroke: getColorSpec(euiTheme, outliers, color, legendType),
+              opacity: {
+                condition: {
+                  value: 1,
+                  test: `(datum['${OUTLIER_SCORE_FIELD}'] >= mlOutlierScoreThreshold.cutoff)`,
+                },
+                value: 0.5,
+              },
+            }
+          : {}),
+        ...(outliers
+          ? {
+              order: { field: OUTLIER_SCORE_FIELD },
+              size: {
+                ...(!dynamicSize
+                  ? {
+                      condition: {
+                        value: 40,
+                        test: `(datum['${OUTLIER_SCORE_FIELD}'] >= mlOutlierScoreThreshold.cutoff)`,
+                      },
+                      value: 8,
+                    }
+                  : {
+                      type: LEGEND_TYPES.QUANTITATIVE,
+                      field: OUTLIER_SCORE_FIELD,
+                      scale: {
+                        type: 'linear',
+                        range: [8, 200],
+                        domain: [0, 1],
+                      },
+                    }),
+              },
+            }
+          : {}),
+        x: {
+          type: LEGEND_TYPES.QUANTITATIVE,
+          field: { repeat: 'column' },
+          scale: { zero: false },
+        },
+        y: {
+          type: LEGEND_TYPES.QUANTITATIVE,
+          field: { repeat: 'row' },
+          scale: { zero: false },
+        },
+        tooltip: [
+          ...columns.map((d) => ({ type: LEGEND_TYPES.QUANTITATIVE, field: d })),
+          ...(outliers
+            ? [{ type: LEGEND_TYPES.QUANTITATIVE, field: OUTLIER_SCORE_FIELD, format: '.3f' }]
+            : []),
+        ],
+      },
+      ...(outliers
+        ? {
+            selection: {
+              mlOutlierScoreThreshold: {
```

Comment:
As discussed, would be nice to retain the threshold setting when the user changes the dynamic size or random scoring switches. Currently it resets to 0.99 whenever you adjust one of those switches.